facenet_sandberg/generate_pairs.py

```diff
@@ -4,139 +4,124 @@
 
 import os
 import random
-from typing import List, Tuple
-
+import sys
+import argparse
 import numpy as np
+from typing import List, Tuple, cast
+
 
-names = [
-    d for d in os.listdir(image_dir) if os.path.isdir(
-        os.path.join(image_dir, d))]
+Mismatch = Tuple[str, int, str, int]
+Match = Tuple[str, int, int]
 
 
-def split_people_into_sets(image_dir: str, k_num_sets: int) -> List[List[str]]:
-    names = [
-        d for d in os.listdir(image_dir) if os.path.isdir(
-            os.path.join(
-                image_dir,
-                d))]
+def write_pairs(fname: str,
+                match_folds: List[List[Match]],
+                mismatch_folds: List[List[Mismatch]],
+                k_num_sets: int,
+                total_matches_mismatches: int) -> None:
+    file_contents = f'{k_num_sets}\t{total_matches_mismatches}\n'
+    for match_fold, mismatch_fold in zip(match_folds, mismatch_folds):
+        for match in match_fold:
+            file_contents += f'{match[0]}\t{match[1]}\t{match[2]}\n'
+        for mismatch in mismatch_fold:
+            file_contents += f'{mismatch[0]}\t{mismatch[1]}\t\
+{mismatch[2]}\t{mismatch[3]}\n'
+    with open(fname, 'w') as fpairs:
```

Comment:
This should specify the encoding (e.g. `encoding='utf-8'`) so that the script doesn't depend on the system default encoding which varies across platforms (e.g. it's usually utf-8 on Linux but not on Windows).